src/DefaultProfile.cpp

```diff
@@ -10,40 +12,24 @@
 #include "geopm_prof.h"
 #include "geopm/Exception.hpp"
 
-#include "config.h"
-
 // Don't allow multithreaded use of non-tprof profile calls
-static __thread int g_pmpi_prof_enabled = 0;
-static int g_pmpi_tprof_enabled = 0;
+static __thread int g_prof_enabled = 0;
 
 namespace geopm
 {
     class DefaultProfile : public ProfileImp
     {
         public:
             DefaultProfile();
-            virtual ~DefaultProfile();
-            void enable_pmpi(void) override;
+            virtual ~DefaultProfile() = default;
     };
 
     DefaultProfile::DefaultProfile()
         : ProfileImp()
     {
-        g_pmpi_prof_enabled = m_is_enabled;
-        g_pmpi_tprof_enabled = m_is_enabled;
-    }
-
-    DefaultProfile::~DefaultProfile()
-    {
-        g_pmpi_prof_enabled = 0;
-        g_pmpi_tprof_enabled = 0;
+        g_prof_enabled = 1;
```

Comment:
Why are you enabling profiling in the destructor?  The seems to be the opposite of the previous behavior.